msgpack-core/src/main/java/io/zeebe/msgpack/spec/MsgPackToken.java

```diff
@@ -64,15 +62,15 @@ public DirectBuffer getValueBuffer() {
   }
 
   public void setValue(DirectBuffer buffer, int offset, int length) {
-    if (offset < buffer.capacity() && offset + length <= buffer.capacity()) {
+    if (length == 0) {
+      valueBuffer.wrap(0, 0);
+    } else if (offset + length <= buffer.capacity()) {
       this.valueBuffer.wrap(buffer, offset, length);
-    } else if (offset == buffer.capacity() && length == 0) {
-      this.valueBuffer.wrap(EMPTY_BYTE);
     } else {
       final int result = offset + length;
       throw new MsgpackReaderException(
           String.format(
-              "Reading %d bytes past buffer capacity(%d) in range [%d:%d)",
+              "Reading %d bytes past buffer capacity(%d) in range [%d:%d]",
```

Comment:
This was intentional more of mathematical notion indicating open edge on right end

Reply:
Ah sorry. I thought it was a typo :sweat_smile: